sources/osgViewer/input/source/InputSourceWebVR.js

```diff
@@ -62,57 +68,78 @@ utils.createPrototypeObject(
-                                    for (i = 0; i < callbacks.length; i++) {
-                                        callback = callbacks[i];
-                                        callback(event);
-                                    }
+            var self = this;
+            navigator.getVRDisplays().then(
+                function(displays) {
+                    if (displays.length > 0) {
+                        if (self._hmd !== displays[0]) {
+                            notify.log('Found a VR display');
+                            //fire the disconnect event
+                            var event = self._events['vrdisplaydisconnected'];
+                            event.vrDisplay = self._hmd;
+                            var i, callback;
+                            var callbacks = self._callbacks['vrdisplaydisconnected'];
+                            if (callbacks) {
+                                for (i = 0; i < callbacks.length; i++) {
+                                    callback = callbacks[i];
+                                    callback(event);
                                 }
+                            }
 
-                                //fire the connect event
-                                event = self._events['vrdisplayconnected'];
-                                event.vrDisplay = displays[0];
-                                callbacks = self._callbacks['vrdisplayconnected'];
-                                if (callbacks) {
-                                    for (i = 0; i < callbacks.length; i++) {
-                                        callback = callbacks[i];
-                                        callback(event);
-                                    }
+                            //fire the connect event
+                            event = self._events['vrdisplayconnected'];
+                            event.vrDisplay = displays[0];
+                            callbacks = self._callbacks['vrdisplayconnected'];
+                            if (callbacks) {
+                                for (i = 0; i < callbacks.length; i++) {
+                                    callback = callbacks[i];
+                                    callback(event);
                                 }
-                                self._hmd = displays[0];
-                                self._frameData = new window.VRFrameData();
                             }
+                            self._hmd = displays[0];
+                            self._frameData = new window.VRFrameData();
                         }
-                    });
-                }.bind(this),
-                this._pollInterval
+                    } else {
+                        this.triggerNotFoundEvent();
+                    }
+                }.bind(this)
             );
         },
 
-        setPollInterval: function (interval) {
+        triggerNotFoundEvent: function() {
+            if (this._pollInterval <= 0) {
```

Comment:
this one is not clear: if _pollInterval is negative, how can we get here ?

otw early return too :)

Reply:
When poll interval is negative, this means that we don't poll automatically but that the user may poll manually using the pollHeadsetMethod.
When this method is called manually, we need to notify the user that no headset has been found when relevant.
However on automatic polling, we don't want to trigger it. Else if would be triggered every pollInterval when no device is plugged.

Reply:
ha, didn't see any pollHeadset manual call anywhere, that's why I'm surprised